src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,358 @@
+    };
+    if (controllerState === ControllerState.READY) {
+      return (
+        <CardHeader style={style}>
+          <CardActions>{actions}</CardActions>
+        </CardHeader>
+      );
+    } else {
+      return (
+        <CardHeader style={style}>
+          <CardTitle>{props.chartKind}</CardTitle>
+          <CardActions>{props.actions}</CardActions>
+        </CardHeader>
+      );
+    }
+  }, [props.actions, props.span, props.chartKind, controllerState, actions]);
+
+  const handleCreateRecording = React.useCallback(() => {
+    history.push({
+      pathname: '/recordings/create',
+      state: {
+        restartExisting: true,
+        name: RECORDING_NAME,
+        templateName: 'Continuous',
+        templateType: 'TARGET',
+        labels: [{ key: 'origin', value: RECORDING_NAME }],
+        duration: -1,
+        // TODO these are arbitrary defaults that will be set in the recording creation form.
+        // Should these values be inferred in some more intelligent way?
+        maxAge: 120, // seconds
+        maxSize: 100 * 1024 * 1024, // bytes
+      } as CreateRecordingProps,
+    });
+  }, [history]);
+
+  return (
+    <DashboardCard
+      id={props.chartKind + '-chart-card'}
+      dashboardId={props.dashboardId}
+      cardSizes={ChartCardSizes}
+      isCompact
+      style={cardStyle}
+      cardHeader={header}
+    >
+      <CardBody>
+        {controllerState === ControllerState.UNKNOWN ? (
+          <LoadingView />
+        ) : controllerState === ControllerState.READY ? (
+          <iframe
+            className="disabled-pointer"
+            key={controllerState + randomKey}
+            style={{ height: '100%', width: '100%' }}
+            src={chartSrc}
+          />
+        ) : (
+          <Bullseye>
+            <EmptyState variant={EmptyStateVariant.large}>
+              <EmptyStateIcon icon={DataSourceIcon} />
+              <Title headingLevel="h2" size="md">
+                {t('CHART_CARD.NO_RECORDING.TITLE')}
+              </Title>
+              <EmptyStateBody>
+                <Trans
+                  t={t}
+                  values={{ recordingName: RECORDING_NAME }}
+                  components={{ label: <Label color="blue" isCompact /> }}
+                >
+                  CHART_CARD.NO_RECORDING.DESCRIPTION
+                </Trans>
+              </EmptyStateBody>
+              <Button variant="primary" onClick={handleCreateRecording}>
+                {t('CHART_CARD.BUTTONS.CREATE.LABEL')}
+              </Button>
+            </EmptyState>
+          </Bullseye>
+        )}
+      </CardBody>
+    </DashboardCard>
+  );
+};
+
+export const ChartCardSizes: DashboardCardSizes = {
+  span: {
+    minimum: 3,
+    default: 4,
+    maximum: 12,
+  },
+  height: {
+    // TODO: implement height resizing
+    minimum: Number.NaN,
+    default: Number.NaN,
+    maximum: Number.NaN,
+  },
+};
+
+export const ChartCardDescriptor: DashboardCardDescriptor = {
```

Comment:
Should all of this be localized as well?

Reply:
Probably, but I thought that made sense to do as a separate refactoring since it applies to these cards as well as the analysis card and the two placeholder cards.